### PR TITLE
Byi metrics http

### PR DESCRIPTION
**Pull request by bin3377**
- Add support for metrics sending
- Add unit test
- promote the version to 1.1.0
- Document updte

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,9 +1,10 @@
 # Logstash Sumo Logic Output Plugin
 
-This is a plugin for [Logstash](https://github.com/elastic/logstash).
+This is an output plugin for [Logstash](https://github.com/elastic/logstash).
 It is fully free and fully open source. The license is Apache 2.0, meaning you are pretty much free to use it however you want in whatever way.
 
 ## Getting Started
+This guide is for the users just want download the binary and make the plugin work. For the developer, please refer to the [Developer Guide](DEVELOPER.md)
```
Comment (rvmiller89) [inline README.md]: "for the users that just want to download the binary"

File: README.md [hunk 2]
```diff
@@ -16,57 +17,67 @@ https://events.sumologic.net/receiver/v1/http/XXXXXXXXXX
...
+(This sample may require installing the [plugins-filters-metrics](https://www.elastic.co/guide/en/logstash/current/plugins-filters-metrics.html) plugin first)
 
+Launch sample with:
+```sh
+bin/logstash -f samples/metrics.conf
 ```
+A mocked event will be sent to Sumo Logic cloud service as 1 minute and 15 minutes rate metrics.
 
+### 6. Get result from Sumo Logic web app
+Logon to Sumo Logic [web app](https://prod-www.sumologic.net/ui/) and run
```
Comment (rvmiller89) [inline README.md]: I think this URL should be `https://service.sumologic.com`  (not prod or .net)

File: lib/logstash/outputs/sumologic.rb [hunk 3]
```diff
@@ -17,35 +18,98 @@ class LogStash::Outputs::SumoLogic < LogStash::Outputs::Base
...
+  # The formatter of log message, by default is message with timestamp and host as prefix
+  # Use %{@json} tag to send whole event
+  config :format, :validate => :string, :default => "%{@timestamp} %{host} %{message}"
 
-  # This lets you choose the structure and parts of the event that are sent in @json tag.
+  # Override the structure of @json tag with the given key value pairs
   config :json_mapping, :validate => :hash
+  
+  # Send metric(s) if configured. This is a hash with k as metric name and v as metric value
+  # Both metric names and values support dynamic strings like %{host}
```
Comment (rvmiller89) [inline lib/logstash/outputs/sumologic.rb]: metric value supports text?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,23 +1,5 @@
-## 1.0.0
- - First public release
-
-### 1.0.1
- - Update gem description 
+## 1.1.0
+ - Support metrics sending
 
-### 1.0.2
- - Support using `%{@json}` in format to send event in json format
```
[the remark this answers is not in the document]
Comment (bin3377) [reply CHANGELOG.md]: The feature is still there. but I want move the details to the readme file. change log just keep the major version

File: lib/logstash/outputs/sumologic.rb (hunk 3 above)
Comment (bin3377) [reply lib/logstash/outputs/sumologic.rb]: use `%{var}` for using the value of variable. the value should be a number, or it will be ignored

**Pull request merged**